File: server/zc/twotieredkanban/site.py
import persistent
from . import model

class Site(persistent.Persistent):

    def __init__(self, initial_email):
        from BTrees.OOBTree import BTree
        self.boards = BTree()
        self.users = self.admins = [initial_email] # Future: persistent

    def update_users(self, users, admins):
        self.users = list(users)
        self.admins = list(admins)
        for board in self.boards.values():
            board.update_users(self.users, self.admins)

    def add_board(self, name, title, description):
        self.boards[name] = model.Kanban(name, title, description)
        self.boards[name].update_users(self.user, self.admins)

File: client/model/site.js
export class Site {

  constructor() {
    this.boards = [];
    this.users = [];
    this.user = {email: ''};
  }

  update(data) {
    if (data.site) {
      Object.assign(this, data.site);
    }
    if (data.user) {
      this.user = data.user;
    }
  }
}
